### PR TITLE
Update ACL syncer: make default deny acls filter more strict,

### PR DESCRIPTION
**Pull request by npinaeva**
delete all acls that were not synced since they should be stale. Add unit test for a specific multicast leftover acl that was a result of libovsdb migration.

Related-to: https://issues.redhat.com/browse/OCPBUGS-21668

### CODE REVIEW

File: go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync_test.go [hunk 1]
```diff
@@ -749,4 +749,91 @@ var _ = ginkgo.Describe("OVN ACL Syncer", func() {
 		}
 		testSyncerWithData(testData, controllerName, []libovsdbtest.TestData{}, nil, nil)
 	})
+	ginkgo.It("deletes leftover multicast acls", func() {
```
Comment (trozet) [inline go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync_test.go:752]: for the test I wish there were more comments explaining exactly what you are creating and the inline comments for the steps on what you are testing. Its hard to follow, and I dont get why the tier is updated, but I trust you did the test case correctly so will merge it.

**Review by trozet — APPROVED** [review]

File: go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync.go [hunk 2]
```diff
@@ -136,6 +136,23 @@ func (syncer *aclSyncer) SyncACLs(existingNodes *v1.NodeList) error {
 		if err != nil {
 			return fmt.Errorf("cannot update stale ACLs: %v", err)
 		}
+
+		// There may be very old acls that are not selected by any of the syncers, delete them.
+		// One example is stale multicast ACLs with the old priority that was accidentally changed by
+		// https://github.com/ovn-org/ovn-kubernetes/commit/f68d302664e64093c867c0b9efe08d1d757d6780#diff-cc83e19af1c257d5a09b711d5977d8f8c20beb34b7b5d3eb37b2f2c53ded1bf7L537-R462
```
Comment (tssurya) [inline go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync.go:142]: cc @arghosh93 : this is the fix i was talking about.
